src/Microsoft.ML.Data/Evaluators/RankingEvaluator.cs

```diff
@@ -948,41 +953,41 @@ private protected override IEnumerable<string> GetPerInstanceColumnsToSave(RoleM
 
     internal static class RankingUtils
     {
-        private static readonly object _lock = new object();
-        private static volatile Double[] _discountMap;
-        private static volatile int _maxTruncationLevel = 0;
+        // Truncation levels are typically less than 100. So we maintain a fixed discount map of size 100
+        // If truncation level greater than 100 is required, we build a new one and return that.
+        private const int FixedDiscountMapSize = 100;
+        private static Double[] _discountMapFixed;
 
-        public static Double[] DiscountMap
+        private static Double[] GetDiscountMapCore(int truncationLevel)
         {
-            get
-            {
-                return _discountMap;
-            }
+            var discountMap = new Double[FixedDiscountMapSize];
```

Comment:
This should use the truncationLevel passed into the function.

Reply:
Oops. Sorry. Fixed with the latest checkin. 

---
In reply to: [333276567](https://github.com/dotnet/machinelearning/pull/4081#discussion_r333276567) [](ancestors = 333276567)